src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -159,7 +159,7 @@ class AppsResource @Inject() (
     * @param body is the raw, unparsed JSON
     * @param updateType CompleteReplacement if we want to replace the app entirely, PartialUpdate if we only want to update provided parts
     */
-  def canonicalAppUpdateFromJson(appId: PathId, body: Array[Byte], updateType: UpdateType): raml.AppUpdate = {
+  def canonicalAppUpdateFromJson(appId: PathId, body: Array[Byte], updateType: UpdateType, existingSecrets: Option[Map[String, Secret]] = None): raml.AppUpdate = {
```

Comment:
What do you think about passing the old app definition instead of just parts of it?

Reply:
➕ 1